lib/hllcounter.cc

```diff
@@ -481,16 +397,16 @@ bool HLLCounter::check_and_normalize_read(std::string &read) const
 {
     bool is_valid = true;
 
-    if (read.length() < this->_ksize) {
+    if (read.length() < _ksize) {
         return false;
     }
 
-    for (unsigned int i = 0; i < read.length(); i++) {
-        read[i] &= 0xdf; // toupper - knock out the "lowercase bit"
-        if (read[i] == 'N') {
-            read[i] = 'A';
+    for (auto &ch: read) {
```

Comment:
as above, should we be checking here and if yes can we use one of the centralised functions to do it?